File: Python/Python Advance Series/Cryptography/CaesarCipher.py
def encrypt(text, key):
    result = ""
    for char in text:
        if char.isalpha():  # Check if the character is a letter
            shift = ord('a') if char.islower() else ord('A')  # Determine the shift value based on lowercase or uppercase
            # Encrypt the character using the standard alphabet series and the key
            encrypted_char = chr((ord(char) - shift + key) % 26 + shift)
            result += encrypted_char  # Append the encrypted character to the result
        else:
            result += char  # Append non-letter characters as they are (e.g., space, punctuation)
    return result  # Return the encrypted text

def decrypt(text, key):
    return encrypt(text, -key)  # Decryption is the same as encryption with the inverse key

def main():
    # Get user input: encryption or decryption
    choice = int(input("Enter 1 for encryption, 2 for decryption: "))

    # Get user input: the text to be encrypted/decrypted
    text = input("Enter the text: ")

    # Get user input: the secret key (a positive integer)
    key = int(input("Enter the key (a positive integer): "))

    if choice == 1:  # If the user chose encryption
        encrypted_text = encrypt(text, key)  # Encrypt the text using the standard alphabet and the key
        print(f"Encrypted text: {encrypted_text}")  # Display the encrypted text
    elif choice == 2:  # If the user chose decryption
        decrypted_text = decrypt(text, key)  # Decrypt the text using the standard alphabet and the key
        print(f"Decrypted text: {decrypted_text}")  # Display the decrypted text
    else:
        print("Invalid choice. Please enter 1 or 2.")  # Display a message for invalid choices

if __name__ == "__main__":
    main()  # Call the main function to start the program

File: Java/Java Advance Series/Cryptography/CaesarCipher.java
import java.util.Scanner;

public class CaesarCipher {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter 1 for encryption, 2 for decryption: ");
        int choice = scanner.nextInt();

        scanner.nextLine(); // Consume the newline character left by nextInt()

        System.out.print("Enter the text: ");
        String text = scanner.nextLine();

        System.out.print("Enter the key (a positive integer): ");
        int key = scanner.nextInt();

        if (choice == 1) {
            String encryptedText = encrypt(text, key);
            System.out.println("Encrypted text: " + encryptedText);
        } else if (choice == 2) {
            String decryptedText = decrypt(text, key);
            System.out.println("Decrypted text: " + decryptedText);
        } else {
            System.out.println("Invalid choice. Please enter 1 or 2.");
        }

        scanner.close();
    }

    public static String encrypt(String text, int key) {
        StringBuilder result = new StringBuilder();
        for (char ch : text.toCharArray()) {
            if (Character.isLetter(ch)) {
                char shift = Character.isLowerCase(ch) ? 'a' : 'A';
                char encryptedChar = (char) ((ch - shift + key) % 26 + shift);
                result.append(encryptedChar);
            } else {
                result.append(ch);
            }
        }
        return result.toString();
    }

    public static String decrypt(String text, int key) {
        return encrypt(text, 26 - key); // Decryption is the same as encryption with the inverse key
    }
}
